upb/def.c

```diff
@@ -171,6 +173,11 @@ typedef enum {
   UPB_DEFTYPE_FIELD_JSONNAME = 2
 } upb_deftype_t;
 
+static upb_deftype_t deftype(upb_value v) {
+  uintptr_t num = (uintptr_t)upb_value_getconstptr(v);
+  return num & 7;
```

Comment:
deftype does & 7, but unpack_def does & 3.  I think named constants would be helpful here

Reply:
Done. (7 crept in from the branch, where the number of deftypes increased. Agreed a constant is better, the bare constant also crept in from the branch. :) )